xds/src/test/java/io/grpc/xds/XdsClientImplTestBase.java

```diff
@@ -3640,6 +3766,39 @@ private  BootstrapInfo buildBootStrap(String serverUri) {
         .build();
   }
 
+  private <T extends ResourceUpdate> DiscoveryRpcCall startResourceWatcher(
+      XdsResourceType<T> type, String name, ResourceWatcher<T> watcher, Executor executor) {
+    FakeClock.TaskFilter timeoutTaskFilter;
+    switch (type.typeName()) {
+      case "LDS":
+        timeoutTaskFilter = LDS_RESOURCE_FETCH_TIMEOUT_TASK_FILTER;
+        xdsClient.watchXdsResource(type, name, watcher, executor);
```

Comment:
All of the cases contain `xdsClient.watchXdsResource(type, name, watcher, executor);` so you can take it out of the switch and have the switch just set `timeoutTaskFilter`.